api/views.py

```diff
@@ -0,0 +1,21 @@
+from rest_framework.response import Response
+from rest_framework.decorators import api_view
+
+from .serializers import MovieSerializer
+from cinema_project.utils import today, next_days
+from base_app.models import Schedule, Movie
+
+
+@api_view(['GET'])
+def currently_playing(request):
+    movies = [schedule.movie for schedule in
+              Schedule.objects.filter(schedule_time__range=[today(), next_days(7)]).order_by('schedule_time')]
```

Comment:
If we need Movie objects, most of the time we want to start the filter from the Movies.objects manager. Try to adjust this query.
Also, the above one might show duplicates.

Reply:
How is this done ? I have been searching about it and found something about GenericRelation. Is this what you are referring to ?

Reply:
Hello! No, I'm just talking about switching from Schedule.objects.filter to Movie.objects.filter(schedule__schedule_time__range=[today(), next_days(7)]).order_by... and using the linked reverse relationships.

Reply:
I hope I got the relationship's name right.

Reply:
Schedule model has movie foreign key, not the other way round, how is it possible to reverse access without set or related name ? They only thing I found was going in the direction of GenericRelation. Please let me know how I could do it this way, as I tried the query suggested above but it does not work

Reply:
Hi! I'm not sure I understand the issue. I just took a look at the Schedule module, because before I just assumed the names and the relationships.

```
class Schedule(models.Model):
    movie = models.ForeignKey(
        Movie, default=None, on_delete=models.SET_DEFAULT, related_name='schedules')

```
From here, we can see that the Schedule model has a ForeignKey to the Movie model. This means multiple Schedule objects can reference the same Movie. Django defines a reverse relationship by default: in this case, this should be schedule_set but because you provided the related_name parameter the related schedule objects can now be accessed as `schedules`.
`Movie.objects.schedules.all()`

So, if we need Movie objects but we need to filter by the related schedule objects the query could be written as: 
`Movie.objects.filter(schedules__schedule_time__range=[today(), next_days(7)]).order_by('schedules__schedule_time')`
Regarding the order_by, sometimes we need to use annotate.

Reply:
OMG, I used just one underline instead of double : / after related name.  Thank you for the detailed explanation though